common/logger/src/libra_logger.rs

```diff
@@ -304,6 +304,38 @@ impl Writer for StderrWriter {
     }
 }
 
+/// A struct for writing logs to a file
+pub struct FileWriter {
+    log_file: std::sync::RwLock<std::fs::File>,
+}
+
+impl FileWriter {
+    pub fn new(log_file: std::path::PathBuf) -> Self {
+        let file = std::fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(log_file)
+            .expect("Unable to open log file");
+        Self {
+            log_file: std::sync::RwLock::new(file),
+        }
+    }
+}
+
+impl Writer for FileWriter {
+    /// Write to file
+    fn write(&self, log: String) {
+        let result = self
+            .log_file
+            .write()
+            .unwrap()
+            .write_all(format!("{}\n", log).as_bytes());
+        if let Err(err) = result {
+            eprintln!("Unable to write to log file: {}", err.to_string());
+        }
```

Comment:
If only the writing on TCP could be this simple!

Reply:
hahaha